src/MOIWrapper.jl

```diff
@@ -0,0 +1,558 @@
+    (LQOI.VecLin, MOI.Nonnegatives),
+    (LQOI.VecLin, MOI.Nonpositives),
+    (LQOI.VecLin, MOI.Zeros)
+]
+
+mutable struct ClpOptimizer <: LQOI.LinQuadOptimizer
+    LQOI.@LinQuadOptimizerBase
+    # env
+    params::Dict{Symbol,Any}
+    ClpOptimizer(::Void) = new()
+end
+
+### Options
+
+# map option name to C function
+const optionmap = Dict(
+   :PrimalTolerance => set_primal_tolerance,
+   :DualTolerance => set_dual_tolerance,
+   :DualObjectiveLimit => set_dual_objective_limit,
+   :MaximumIterations => set_maximum_iterations,
+   :MaximumSeconds => set_maximum_seconds,
+   :LogLevel => set_log_level,
+   :Scaling => scaling,
+   :Perturbation => set_perturbation,
+   #:Algorithm => set_algorithm
+   )
+# These options are set by using the ClpSolve object
+const solveoptionmap = Dict(
+   :PresolveType => set_presolve_type,
+   :SolveType => set_solve_type,
+   :InfeasibleReturn => set_infeasible_return,
+   )
+
+function setoption(m::ClpOptimizer, name::Symbol, value)
+    if haskey(optionmap, name)
+        optionmap[name](m.inner,value)
+    elseif haskey(solveoptionmap, name)
+        solveoptionmap[name](m.solveroptions,value)
+    else
+        error("Unrecognized option: $name")
+    end
+end
+
+function ClpOptimizer(;kwargs...)
+    # env = Env()
+    m = ClpOptimizer(nothing)
+    # m.env = env
+    m.params = Dict{String,Any}()
+    MOI.empty!(m)
+    for (name,value) in kwargs
+        m.params[Symbol(name)] = value
+    end
+    return m
+end
+
+LQOI.LinearQuadraticModel(::Type{ClpOptimizer},env) = ClpModel() 
+
+LQOI.supported_constraints(s::ClpOptimizer) = SUPPORTED_CONSTRAINTS
+LQOI.supported_objectives(s::ClpOptimizer) = SUPPORTED_OBJECTIVES
+
+"""
+Change the bounds of the variable. The sense of the upperbound
+is given by `backend_type(m, Val{:Upperbound}())`. The sense
+of the lowerbound is given by `backend_type(m, Val{:Lowerbound}())`
+"""
+function LQOI.change_variable_bounds!(instance::ClpOptimizer, cols::Vector{Int}, values::Vector{Float64}, 
+        senses::Vector)
+    
+    upperbounds = replaceInf(get_col_upper(instance.inner))    
+    lowerbounds = replaceInf(get_col_lower(instance.inner))    
+    
+    for i in 1:length(senses)
+        if senses[i] == Cchar('U')
+            upperbounds[cols[i]] = values[i]
+        elseif senses[i] == Cchar('L')
+            lowerbounds[cols[i]] = values[i]
+        else            
+            error("sense is " * string(senses[i]) * ", but only " * Cchar('U') *  
+            " and " * Cchar('L') * " senses are supported")
+        end
+    end
+    
+    chg_column_upper(instance.inner, upperbounds)
+    chg_column_lower(instance.inner, lowerbounds)
+end
+
+"""
+get_variable_lowerbound(m, col::Int)::Float64
+
+Get the lower bound of the variable in 1-indexed column `col` of the model `m`.
+"""
+function LQOI.get_variable_lowerbound(instance::ClpOptimizer, col::Int)::Float64
+    lower = replaceInf(get_col_lower(instance.inner))
+    return lower[col];  
```

Comment:
Two comments. Since you only query one element, you don't need to call `replaceInf` on the whole lowerbound vector. But also, if `get_col_lower` returns `10+20`, why should we change that to `Inf`? The real bound in Clp is `10+20`.